src/backend/cgobj.c

```diff
@@ -979,6 +979,55 @@ void Obj::linnum(Srcpos srcpos,targ_size_t offset)
     }
 }
 
+#define FROMWORD(ptr)	(*(unsigned short*)(ptr))
+#define FROMLONG(ptr)	(*(unsigned long*)(ptr))
+
+/***************************
+ * get the byte offset of a line of code in its segment
+ * if no debug info for the exact line is found, return the offset
+ *  of the closest line with debug info before the given line
+ */
+targ_size_t Obj::getlineoffset(Srcpos srcpos)
```

Comment:
I noticed this is incomplete because there is an optimization from the 80s in Obj::linnum that flushes and forgets line number info for the "main" object file every 512 bytes.